src/v2i-hub/PedestrianPlugin/src/include/PedestrianPlugin.hpp

```diff
@@ -62,51 +51,56 @@ namespace PedestrianPlugin
 class PedestrianPlugin: public PluginClient
 {
 public:
-	PedestrianPlugin(std::string);
-	PedestrianPlugin();
-	virtual ~PedestrianPlugin();
-	int Main();
-	uint16_t webport;
-	std::string webip; 
-	std::string webSocketIP;
-	std::string webSocketURLExt;
-	std::string dataprovider;
-	float cameraRotation;
-	std::shared_ptr<FLIRWebSockAsyncClnSession> flirSession;
-	std::string hostString;
+	explicit PedestrianPlugin(const std::string &name);
+	int Main() override;
 
 protected:
 	void UpdateConfigSettings();
 
 	// Virtual method overrides.
-	void OnConfigChanged(const char *key, const char *value);
-	void OnStateChange(IvpPluginState state);
+	void OnConfigChanged(const char *key, const char *value) override;
+	void OnStateChange(IvpPluginState state) override;
 
-	void HandleMapDataMessage(MapDataMessage &msg, routeable_message &routeableMsg);
-	void HandleBasicSafetyMessage(BsmMessage &msg, routeable_message &routeableMsg);
-	void BroadcastPsm(char *psmJson);
+	void BroadcastPsm(const std::string &psmJson);
 
 	int  StartWebService();
+	int  StartWebSocket();
+	void StopWebService();
+	void StopWebSocket();
 	void PedestrianRequestHandler(QHttpEngine::Socket *socket);
-	void writeResponse(int responseCode , QHttpEngine::Socket *socket);
-
-	int StartWebSocket();
+	void writeResponse(int responseCode , QHttpEngine::Socket *socket) const;
 
 	void OnWebSocketConnected();
 	void OnWebSocketDataReceived(QString message);
 	void OnWebSocketClosed();
-	
-	int checkXML();
-
-
 
+	void checkXML();
+	
 private:
-	tmx::utils::UdpClient *_signSimClient = NULL;
+	std::unique_ptr<tmx::utils::UdpClient> _signSimClient = nullptr;
+	
 	J2735MessageFactory factory;
 	
+	std::mutex _cfgLock;
+	
+	uint16_t webport;
+	std::string webip; 
+	std::string webSocketIP;
+	std::string webSocketURLExt;
+	int instance;
+	std::string dataprovider;
+	float cameraRotation;
+	std::shared_ptr<FLIRWebSockAsyncClnSession> flirSession;
+	std::string hostString;
 
-};
-std::mutex _cfgLock;
+	bool runningWebSocket = false;
+    bool runningWebService = false;
+
+	// The io_context is required for all I/O
+    net::io_context ioc;
 
+	// API URL to accept PSM XML
+	const QString PSM_Receive = "";
```

Comment:
This is never updated. Is this intentional. We should either set a fixed value for the endpoint or not have it as a const

Reply:
Yeah, this QObjectHandler requires a URL extension, but we don't want to add anything because we're using POST. For example, we want to keep 127.0.0.1:9000/ instead of 127.0.0.1:9000/someExtension.

Would you prefer to have empty quotes in the registerMethod or use this empty const? Thought this looked better, but it isn't needed.

Reply:
Would we break anything if we required some extension. It is probably better practice like /psm or /pedestriansafetymessage